### DISCUSSION TITLE
[i18n] Add option to disable default fallback locale in path

### DISCUSSION THREAD

**Discussion by rokinsky** [vercel/next.js#19097] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

When I use the `i18n` feature with a catch-all route, I cannot determine if I'm in a locale-fallback state, i.e. `/en/about` or `/about` (assuming `en` is the default locale).

## Describe the solution you'd like

With following config
```js
module.exports = {
  i18n: {
    locales: [
      "en",
      "nl-nl",
      "de-de",
      // ...
    ],
    defaultLocale: "en",
    defaultFallback: false,
  },
};
```

and with `pages/about.jsx`  and `pages/index.jsx` routes,

when visiting
* `/en/about` should work as expected
* `/about` I would rather see a 404 error page or redirect to `/en/about`
* `/` I would prefer to be redirected to `/en` (using existing strategy).

Additionally, it means using `{ locale: false }` in `next/router` and `next/link` by default.

## Describe alternatives you've considered

* Do not use the built-in Next.js i18n feature and prefer the `pages/[locale]/*` approach instead.
* Add to gSSP/gSP `context` and `next/router` variable `isLocaleFallback: boolean`

## Additional context

N/A

**Comment by multivoltage** [comment]
please say me there is a method to do that

**Reply by rokinsky** [reply]
You can try the following workaround: https://github.com/vercel/next.js/discussions/18419#discussioncomment-327128

**Reply by gregghawes** [reply]
Doesn't that force a subfolder though? Did you manage to do it without a default subfolder?

**Reply by multivoltage** [reply]
After 1 year we have a production site for my company with:
- site.com/en/about-us
- site.com/chi-siamo (chi siamo is equivalent of about-us in italian).

I have about.us.tsx inside en folder and chi-siamo.tsx inside root. Both files render an AboutPages.tsx, a custom component inside src/abstractPahes folder.

I use next-i18next only to call useTranslation. I did not setup i18n node inside next config.
I need to pass differents locales in about-us.tsx and chi-siamo.tsx when render <AboutPages locale="en|it"/>

**Comment by raphjatteau** [comment]
This can be quite important for SEO indeed.
@rokinsky as it is not built in next.js, have you found some ideas that can work well as a workaround ?

**Reply by rokinsky** [reply]
You can try the following workaround: https://github.com/vercel/next.js/discussions/18419#discussioncomment-327128
